pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -388,6 +396,84 @@ func (m *topologyPairsMaps) clone() *topologyPairsMaps {
 	return copy
 }
 
+func (podSpreadMap *topologyPairsPodSpreadMap) addPod(addedPod, metapod *v1.Pod, node *v1.Node) error {
```

Comment:
I think add and remove can be simplified and made faster if we just keep track of which exact topologyPair had the minimum match, combined with the suggestion I made [in the second PR](https://github.com/kubernetes/kubernetes/pull/77760#issuecomment-512516312).

Reply:
+1. For the case where there are several nodes with the minimum value, update only one instead of all of them.

Although, the most expensive for seems to be this `for pair, podSet := range podSpreadMap.topologyPairToPods`. Why don't we have a `map[topologyKey][value]int?`

Reply:
> I think "metapod" has a different meaning than the one we intend for this parameter, perhaps targetPod?

Maybe `preemptorPod`.

Reply:
> I think add and remove can be simplified and made faster if we just keep track of which exact topologyPair had the minimum match, combined with the suggestion I made [in the second PR](https://github.com/kubernetes/kubernetes/pull/77760#issuecomment-512516312).

I will raise another PR to optimize the data structure of `podSpreadMap`.

Reply:
SG for both.